{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "16531a5d-91b3-45cb-a6b4-ac8644257188",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: igraph\n",
      "\n",
      "\n",
      "Attaching package: ‘igraph’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    decompose, spectrum\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "Loading required package: AnnotationHub\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    normalize, path, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: BiocFileCache\n",
      "\n",
      "Loading required package: dbplyr\n",
      "\n",
      "Loading required package: MultiAssayExperiment\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationHub’:\n",
      "\n",
      "    cache\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.18.0\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite either one:\n",
      "- Gu, Z. Complex Heatmap Visualization. iMeta 2022.\n",
      "- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "    genomic data. Bioinformatics 2016.\n",
      "\n",
      "\n",
      "The new InteractiveComplexHeatmap package can directly export static \n",
      "complex heatmaps into an interactive Shiny app with zero effort. Have a try!\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "\n",
      "\n",
      "========================================\n",
      "circlize version 0.4.15\n",
      "CRAN page: https://cran.r-project.org/package=circlize\n",
      "Github page: https://github.com/jokergoo/circlize\n",
      "Documentation: https://jokergoo.github.io/circlize_book/book/\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. circlize implements and enhances circular visualization\n",
      "  in R. Bioinformatics 2014.\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(circlize))\n",
      "========================================\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘circlize’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    degree\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘webshot’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    resize\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    resize\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    resize\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dbplyr’:\n",
      "\n",
      "    ident, sql\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    as_data_frame, groups, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(hipathia)\n",
    "library(AnnotationDbi)\n",
    "library(org.Hs.eg.db)\n",
    "library(ComplexHeatmap)\n",
    "library(circlize)\n",
    "library(ggplot2)\n",
    "library(jsonlite)\n",
    "library(networkD3)\n",
    "library(webshot)\n",
    "library(dplyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "079ecc2e-d0f5-4f26-82d7-5adc3c8d7df5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loaded 146 pathways\n",
      "\n"
     ]
    }
   ],
   "source": [
    "pathways <- load_pathways(\"hsa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d370141c-0970-4b13-a455-81b57e92d10e",
   "metadata": {},
   "outputs": [],
   "source": [
    "ko_sign_files <- list.files(\"/mnt/lustre/scratch/CBRA/research/projects/rrdd_mm/results/05_clasi_result_kstest/05_1_clasi_result_kstest_sign/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "63ff6180-37e5-4524-87b5-f093d2cd2d6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ko_kstest <-do.call(rbind,lapply(ko_sign_files, FUN=function(x) readRDS(paste(\"/mnt/lustre/scratch/CBRA/research/projects/rrdd_mm/results/05_clasi_result_kstest/05_1_clasi_result_kstest_sign/\",x,sep=\"\"))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d818061c-8f01-46dc-95e1-a9142f18ae29",
   "metadata": {},
   "outputs": [],
   "source": [
    "ko_kstest_sign <- ko_kstest[ko_kstest$kspvadj < 0.05,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "aa9923ff-21da-46c4-84a7-324fb5b2e1fc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned many:1 mapping between keys and columns\n",
      "\n"
     ]
    }
   ],
   "source": [
    "name_enfermerdad_symbol <- readRDS(\"/mnt/lustre/scratch/CBRA/research/projects/rrdd_mm/results/00_parser_orphanet/gen_enfermedad/name_enfermerdad_symbol.rds\")\n",
    "colnames(name_enfermerdad_symbol) <- c(\"name_enf\",\"gene\")\n",
    "symbol_entreid <- AnnotationDbi::select(org.Hs.eg.db, keys=paste(name_enfermerdad_symbol$gene), columns=c(\"ENTREZID\"),keytype=\"SYMBOL\")\n",
    "\n",
    "gen_enf <- unique(merge(name_enfermerdad_symbol, symbol_entreid, by.x=\"gene\",by.y=\"SYMBOL\"))\n",
    "rd_gen_class <- readRDS(\"/mnt/lustre/scratch/CBRA/research/projects/rrdd_mm/results/00_parser_orphanet/rd_gen_class.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1a005e7d-9bdd-4b11-a193-9d381fe1ee3f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>name_enf</th><th scope=col>ENTREZID</th><th scope=col>clasificacion</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AARS </td><td>Autosomal dominant Charcot-Marie-Tooth disease type 2N                         </td><td>NA   </td><td>Rare neurologic disease         </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AARS2</td><td>Combined oxidative phosphorylation defect type 8                               </td><td>57505</td><td>Rare inborn errors of metabolism</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AARS2</td><td>Hereditary diffuse leukoencephalopathy with axonal spheroids and pigmented glia</td><td>57505</td><td>Rare neurologic disease         </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AASS </td><td>Hyperlysinemia                                                                 </td><td>10157</td><td>Rare inborn errors of metabolism</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AASS </td><td>Saccharopinuria                                                                </td><td>10157</td><td>Rare inborn errors of metabolism</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ABAT </td><td>Gamma-aminobutyric acid transaminase deficiency                                </td><td>18   </td><td>Rare inborn errors of metabolism</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & gene & name\\_enf & ENTREZID & clasificacion\\\\\n",
       "  & <fct> & <fct> & <chr> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & AARS  & Autosomal dominant Charcot-Marie-Tooth disease type 2N                          & NA    & Rare neurologic disease         \\\\\n",
       "\t2 & AARS2 & Combined oxidative phosphorylation defect type 8                                & 57505 & Rare inborn errors of metabolism\\\\\n",
       "\t3 & AARS2 & Hereditary diffuse leukoencephalopathy with axonal spheroids and pigmented glia & 57505 & Rare neurologic disease         \\\\\n",
       "\t4 & AASS  & Hyperlysinemia                                                                  & 10157 & Rare inborn errors of metabolism\\\\\n",
       "\t5 & AASS  & Saccharopinuria                                                                 & 10157 & Rare inborn errors of metabolism\\\\\n",
       "\t6 & ABAT  & Gamma-aminobutyric acid transaminase deficiency                                 & 18    & Rare inborn errors of metabolism\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | gene &lt;fct&gt; | name_enf &lt;fct&gt; | ENTREZID &lt;chr&gt; | clasificacion &lt;fct&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | AARS  | Autosomal dominant Charcot-Marie-Tooth disease type 2N                          | NA    | Rare neurologic disease          |\n",
       "| 2 | AARS2 | Combined oxidative phosphorylation defect type 8                                | 57505 | Rare inborn errors of metabolism |\n",
       "| 3 | AARS2 | Hereditary diffuse leukoencephalopathy with axonal spheroids and pigmented glia | 57505 | Rare neurologic disease          |\n",
       "| 4 | AASS  | Hyperlysinemia                                                                  | 10157 | Rare inborn errors of metabolism |\n",
       "| 5 | AASS  | Saccharopinuria                                                                 | 10157 | Rare inborn errors of metabolism |\n",
       "| 6 | ABAT  | Gamma-aminobutyric acid transaminase deficiency                                 | 18    | Rare inborn errors of metabolism |\n",
       "\n"
      ],
      "text/plain": [
       "  gene \n",
       "1 AARS \n",
       "2 AARS2\n",
       "3 AARS2\n",
       "4 AASS \n",
       "5 AASS \n",
       "6 ABAT \n",
       "  name_enf                                                                       \n",
       "1 Autosomal dominant Charcot-Marie-Tooth disease type 2N                         \n",
       "2 Combined oxidative phosphorylation defect type 8                               \n",
       "3 Hereditary diffuse leukoencephalopathy with axonal spheroids and pigmented glia\n",
       "4 Hyperlysinemia                                                                 \n",
       "5 Saccharopinuria                                                                \n",
       "6 Gamma-aminobutyric acid transaminase deficiency                                \n",
       "  ENTREZID clasificacion                   \n",
       "1 NA       Rare neurologic disease         \n",
       "2 57505    Rare inborn errors of metabolism\n",
       "3 57505    Rare neurologic disease         \n",
       "4 10157    Rare inborn errors of metabolism\n",
       "5 10157    Rare inborn errors of metabolism\n",
       "6 18       Rare inborn errors of metabolism"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "inf_rrdd <- merge(gen_enf, rd_gen_class, by.x=c(\"gene\",\"name_enf\"), by.y=c(\"V1\",\"enfermedad\"))\n",
    "head(inf_rrdd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d43993ff-d51c-4607-ba83-893f192d7e81",
   "metadata": {},
   "outputs": [],
   "source": [
    "ko_kstest_sign$name_circ <- get_path_names(pathways, ko_kstest_sign$circ)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3bee02a6-1a69-4b35-9a49-597a965a145e",
   "metadata": {},
   "outputs": [],
   "source": [
    "ko_kstest_sign$path <- unlist(lapply(strsplit(ko_kstest_sign$name_circ, \":\", fixed = T), FUN=function(x) x[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2691cd80-6f0d-41c5-a5bf-a074f5af135c",
   "metadata": {},
   "outputs": [],
   "source": [
    "annot_ko_kstest_sign <- merge(inf_rrdd, ko_kstest_sign, by.x=\"ENTREZID\", by.y=\"ko\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "2bee086f-35a1-4bf6-8a78-9d6ea982ea24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ENTREZID</th><th scope=col>gene</th><th scope=col>name_enf</th><th scope=col>clasificacion</th><th scope=col>kspvadj</th><th scope=col>logFC</th><th scope=col>circ</th><th scope=col>sex</th><th scope=col>tissue</th><th scope=col>name_circ</th><th scope=col>path</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>100506658</td><td>OCLN</td><td>Congenital intrauterine infection-like syndrome</td><td>Rare neurologic disease</td><td>0.000000e+00</td><td>-6.643856</td><td>P-hsa04530-4 53   </td><td>male  </td><td>Skin          </td><td>Tight junction: OCLN OCLN     </td><td>Tight junction</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>100506658</td><td>OCLN</td><td>Congenital intrauterine infection-like syndrome</td><td>Rare neurologic disease</td><td>7.507341e-30</td><td>-6.643856</td><td>P-hsa04530-4 45   </td><td>male  </td><td>Nerve         </td><td>Tight junction: OCLN TJP2     </td><td>Tight junction</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>100506658</td><td>OCLN</td><td>Congenital intrauterine infection-like syndrome</td><td>Rare neurologic disease</td><td>1.289345e-08</td><td>-6.643856</td><td>P-hsa04530-4 37 11</td><td>male  </td><td>Salivary Gland</td><td>Tight junction: OCLN TJP3 TJP1</td><td>Tight junction</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>100506658</td><td>OCLN</td><td>Congenital intrauterine infection-like syndrome</td><td>Rare neurologic disease</td><td>7.910729e-13</td><td>-6.561009</td><td>P-hsa04530-4 37 11</td><td>female</td><td>Muscle        </td><td>Tight junction: OCLN TJP3 TJP1</td><td>Tight junction</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>100506658</td><td>OCLN</td><td>Congenital intrauterine infection-like syndrome</td><td>Rare neurologic disease</td><td>1.982331e-29</td><td>-6.643856</td><td>P-hsa04530-4 53   </td><td>male  </td><td>Thyroid       </td><td>Tight junction: OCLN OCLN     </td><td>Tight junction</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>100506658</td><td>OCLN</td><td>Congenital intrauterine infection-like syndrome</td><td>Rare neurologic disease</td><td>1.860340e-23</td><td>-6.643856</td><td>P-hsa04530-4 53   </td><td>female</td><td>Adipose Tissue</td><td>Tight junction: OCLN OCLN     </td><td>Tight junction</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ENTREZID & gene & name\\_enf & clasificacion & kspvadj & logFC & circ & sex & tissue & name\\_circ & path\\\\\n",
       "  & <chr> & <fct> & <fct> & <fct> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & 100506658 & OCLN & Congenital intrauterine infection-like syndrome & Rare neurologic disease & 0.000000e+00 & -6.643856 & P-hsa04530-4 53    & male   & Skin           & Tight junction: OCLN OCLN      & Tight junction\\\\\n",
       "\t2 & 100506658 & OCLN & Congenital intrauterine infection-like syndrome & Rare neurologic disease & 7.507341e-30 & -6.643856 & P-hsa04530-4 45    & male   & Nerve          & Tight junction: OCLN TJP2      & Tight junction\\\\\n",
       "\t3 & 100506658 & OCLN & Congenital intrauterine infection-like syndrome & Rare neurologic disease & 1.289345e-08 & -6.643856 & P-hsa04530-4 37 11 & male   & Salivary Gland & Tight junction: OCLN TJP3 TJP1 & Tight junction\\\\\n",
       "\t4 & 100506658 & OCLN & Congenital intrauterine infection-like syndrome & Rare neurologic disease & 7.910729e-13 & -6.561009 & P-hsa04530-4 37 11 & female & Muscle         & Tight junction: OCLN TJP3 TJP1 & Tight junction\\\\\n",
       "\t5 & 100506658 & OCLN & Congenital intrauterine infection-like syndrome & Rare neurologic disease & 1.982331e-29 & -6.643856 & P-hsa04530-4 53    & male   & Thyroid        & Tight junction: OCLN OCLN      & Tight junction\\\\\n",
       "\t6 & 100506658 & OCLN & Congenital intrauterine infection-like syndrome & Rare neurologic disease & 1.860340e-23 & -6.643856 & P-hsa04530-4 53    & female & Adipose Tissue & Tight junction: OCLN OCLN      & Tight junction\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 11\n",
       "\n",
       "| <!--/--> | ENTREZID &lt;chr&gt; | gene &lt;fct&gt; | name_enf &lt;fct&gt; | clasificacion &lt;fct&gt; | kspvadj &lt;dbl&gt; | logFC &lt;dbl&gt; | circ &lt;chr&gt; | sex &lt;chr&gt; | tissue &lt;chr&gt; | name_circ &lt;chr&gt; | path &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 100506658 | OCLN | Congenital intrauterine infection-like syndrome | Rare neurologic disease | 0.000000e+00 | -6.643856 | P-hsa04530-4 53    | male   | Skin           | Tight junction: OCLN OCLN      | Tight junction |\n",
       "| 2 | 100506658 | OCLN | Congenital intrauterine infection-like syndrome | Rare neurologic disease | 7.507341e-30 | -6.643856 | P-hsa04530-4 45    | male   | Nerve          | Tight junction: OCLN TJP2      | Tight junction |\n",
       "| 3 | 100506658 | OCLN | Congenital intrauterine infection-like syndrome | Rare neurologic disease | 1.289345e-08 | -6.643856 | P-hsa04530-4 37 11 | male   | Salivary Gland | Tight junction: OCLN TJP3 TJP1 | Tight junction |\n",
       "| 4 | 100506658 | OCLN | Congenital intrauterine infection-like syndrome | Rare neurologic disease | 7.910729e-13 | -6.561009 | P-hsa04530-4 37 11 | female | Muscle         | Tight junction: OCLN TJP3 TJP1 | Tight junction |\n",
       "| 5 | 100506658 | OCLN | Congenital intrauterine infection-like syndrome | Rare neurologic disease | 1.982331e-29 | -6.643856 | P-hsa04530-4 53    | male   | Thyroid        | Tight junction: OCLN OCLN      | Tight junction |\n",
       "| 6 | 100506658 | OCLN | Congenital intrauterine infection-like syndrome | Rare neurologic disease | 1.860340e-23 | -6.643856 | P-hsa04530-4 53    | female | Adipose Tissue | Tight junction: OCLN OCLN      | Tight junction |\n",
       "\n"
      ],
      "text/plain": [
       "  ENTREZID  gene name_enf                                       \n",
       "1 100506658 OCLN Congenital intrauterine infection-like syndrome\n",
       "2 100506658 OCLN Congenital intrauterine infection-like syndrome\n",
       "3 100506658 OCLN Congenital intrauterine infection-like syndrome\n",
       "4 100506658 OCLN Congenital intrauterine infection-like syndrome\n",
       "5 100506658 OCLN Congenital intrauterine infection-like syndrome\n",
       "6 100506658 OCLN Congenital intrauterine infection-like syndrome\n",
       "  clasificacion           kspvadj      logFC     circ               sex   \n",
       "1 Rare neurologic disease 0.000000e+00 -6.643856 P-hsa04530-4 53    male  \n",
       "2 Rare neurologic disease 7.507341e-30 -6.643856 P-hsa04530-4 45    male  \n",
       "3 Rare neurologic disease 1.289345e-08 -6.643856 P-hsa04530-4 37 11 male  \n",
       "4 Rare neurologic disease 7.910729e-13 -6.561009 P-hsa04530-4 37 11 female\n",
       "5 Rare neurologic disease 1.982331e-29 -6.643856 P-hsa04530-4 53    male  \n",
       "6 Rare neurologic disease 1.860340e-23 -6.643856 P-hsa04530-4 53    female\n",
       "  tissue         name_circ                      path          \n",
       "1 Skin           Tight junction: OCLN OCLN      Tight junction\n",
       "2 Nerve          Tight junction: OCLN TJP2      Tight junction\n",
       "3 Salivary Gland Tight junction: OCLN TJP3 TJP1 Tight junction\n",
       "4 Muscle         Tight junction: OCLN TJP3 TJP1 Tight junction\n",
       "5 Thyroid        Tight junction: OCLN OCLN      Tight junction\n",
       "6 Adipose Tissue Tight junction: OCLN OCLN      Tight junction"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(annot_ko_kstest_sign)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "79a2f7e3-4b5e-4235-b706-12913d6f62c6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'sex', 'tissue'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A grouped_df: 33 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>sex</th><th scope=col>tissue</th><th scope=col>path</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>female</td><td>Adipose Tissue</td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Blood Vessel  </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Brain         </td><td>Complement and coagulation cascades</td><td>42</td></tr>\n",
       "\t<tr><td>female</td><td>Brain         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Breast        </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Colon         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Esophagus     </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Heart         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Lung          </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Nerve         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Skin          </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Stomach       </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>female</td><td>Thyroid       </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Adipose Tissue</td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Adrenal Gland </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Blood Vessel  </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Brain         </td><td>Complement and coagulation cascades</td><td>47</td></tr>\n",
       "\t<tr><td>male  </td><td>Brain         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Breast        </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Colon         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Esophagus     </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Heart         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Liver         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Lung          </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Muscle        </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Nerve         </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Pancreas      </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Pituitary     </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Salivary Gland</td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Skin          </td><td>Complement and coagulation cascades</td><td>36</td></tr>\n",
       "\t<tr><td>male  </td><td>Skin          </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Stomach       </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "\t<tr><td>male  </td><td>Thyroid       </td><td>HIF-1 signaling pathway            </td><td>34</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A grouped\\_df: 33 × 4\n",
       "\\begin{tabular}{llll}\n",
       " sex & tissue & path & n\\\\\n",
       " <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t female & Adipose Tissue & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Blood Vessel   & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Brain          & Complement and coagulation cascades & 42\\\\\n",
       "\t female & Brain          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Breast         & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Colon          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Esophagus      & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Heart          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Lung           & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Nerve          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Skin           & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Stomach        & HIF-1 signaling pathway             & 34\\\\\n",
       "\t female & Thyroid        & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Adipose Tissue & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Adrenal Gland  & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Blood Vessel   & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Brain          & Complement and coagulation cascades & 47\\\\\n",
       "\t male   & Brain          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Breast         & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Colon          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Esophagus      & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Heart          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Liver          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Lung           & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Muscle         & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Nerve          & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Pancreas       & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Pituitary      & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Salivary Gland & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Skin           & Complement and coagulation cascades & 36\\\\\n",
       "\t male   & Skin           & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Stomach        & HIF-1 signaling pathway             & 34\\\\\n",
       "\t male   & Thyroid        & HIF-1 signaling pathway             & 34\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A grouped_df: 33 × 4\n",
       "\n",
       "| sex &lt;chr&gt; | tissue &lt;chr&gt; | path &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| female | Adipose Tissue | HIF-1 signaling pathway             | 34 |\n",
       "| female | Blood Vessel   | HIF-1 signaling pathway             | 34 |\n",
       "| female | Brain          | Complement and coagulation cascades | 42 |\n",
       "| female | Brain          | HIF-1 signaling pathway             | 34 |\n",
       "| female | Breast         | HIF-1 signaling pathway             | 34 |\n",
       "| female | Colon          | HIF-1 signaling pathway             | 34 |\n",
       "| female | Esophagus      | HIF-1 signaling pathway             | 34 |\n",
       "| female | Heart          | HIF-1 signaling pathway             | 34 |\n",
       "| female | Lung           | HIF-1 signaling pathway             | 34 |\n",
       "| female | Nerve          | HIF-1 signaling pathway             | 34 |\n",
       "| female | Skin           | HIF-1 signaling pathway             | 34 |\n",
       "| female | Stomach        | HIF-1 signaling pathway             | 34 |\n",
       "| female | Thyroid        | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Adipose Tissue | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Adrenal Gland  | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Blood Vessel   | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Brain          | Complement and coagulation cascades | 47 |\n",
       "| male   | Brain          | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Breast         | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Colon          | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Esophagus      | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Heart          | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Liver          | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Lung           | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Muscle         | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Nerve          | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Pancreas       | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Pituitary      | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Salivary Gland | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Skin           | Complement and coagulation cascades | 36 |\n",
       "| male   | Skin           | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Stomach        | HIF-1 signaling pathway             | 34 |\n",
       "| male   | Thyroid        | HIF-1 signaling pathway             | 34 |\n",
       "\n"
      ],
      "text/plain": [
       "   sex    tissue         path                                n \n",
       "1  female Adipose Tissue HIF-1 signaling pathway             34\n",
       "2  female Blood Vessel   HIF-1 signaling pathway             34\n",
       "3  female Brain          Complement and coagulation cascades 42\n",
       "4  female Brain          HIF-1 signaling pathway             34\n",
       "5  female Breast         HIF-1 signaling pathway             34\n",
       "6  female Colon          HIF-1 signaling pathway             34\n",
       "7  female Esophagus      HIF-1 signaling pathway             34\n",
       "8  female Heart          HIF-1 signaling pathway             34\n",
       "9  female Lung           HIF-1 signaling pathway             34\n",
       "10 female Nerve          HIF-1 signaling pathway             34\n",
       "11 female Skin           HIF-1 signaling pathway             34\n",
       "12 female Stomach        HIF-1 signaling pathway             34\n",
       "13 female Thyroid        HIF-1 signaling pathway             34\n",
       "14 male   Adipose Tissue HIF-1 signaling pathway             34\n",
       "15 male   Adrenal Gland  HIF-1 signaling pathway             34\n",
       "16 male   Blood Vessel   HIF-1 signaling pathway             34\n",
       "17 male   Brain          Complement and coagulation cascades 47\n",
       "18 male   Brain          HIF-1 signaling pathway             34\n",
       "19 male   Breast         HIF-1 signaling pathway             34\n",
       "20 male   Colon          HIF-1 signaling pathway             34\n",
       "21 male   Esophagus      HIF-1 signaling pathway             34\n",
       "22 male   Heart          HIF-1 signaling pathway             34\n",
       "23 male   Liver          HIF-1 signaling pathway             34\n",
       "24 male   Lung           HIF-1 signaling pathway             34\n",
       "25 male   Muscle         HIF-1 signaling pathway             34\n",
       "26 male   Nerve          HIF-1 signaling pathway             34\n",
       "27 male   Pancreas       HIF-1 signaling pathway             34\n",
       "28 male   Pituitary      HIF-1 signaling pathway             34\n",
       "29 male   Salivary Gland HIF-1 signaling pathway             34\n",
       "30 male   Skin           Complement and coagulation cascades 36\n",
       "31 male   Skin           HIF-1 signaling pathway             34\n",
       "32 male   Stomach        HIF-1 signaling pathway             34\n",
       "33 male   Thyroid        HIF-1 signaling pathway             34"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "annot_ko_kstest_sign[annot_ko_kstest_sign$clasificacion == \"Rare circulatory system disease\" | \n",
    "annot_ko_kstest_sign$clasificacion == \"Rare hematologic disease\" |\n",
    "annot_ko_kstest_sign$clasificacion == \"Rare circulatory system disease\" |\n",
    "annot_ko_kstest_sign$clasificacion == \"Rare cardiac disease\",] %>% group_by(sex,tissue, path) %>%\n",
    "summarise(n=n()) %>% filter(n>33)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e4b13c90-1373-411a-8dcc-d7428539ee67",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'clasificacion', 'sex', 'tissue'. You can override using the `.groups` argument.\n",
      "\u001b[1m\u001b[22m`summarise()` has grouped output by 'clasificacion', 'sex'. You can override using the `.groups` argument.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 4 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>clasificacion</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Rare developmental defect during embryogenesis</td><td>85</td></tr>\n",
       "\t<tr><td>Rare immune disease                           </td><td>57</td></tr>\n",
       "\t<tr><td>Rare neoplastic disease                       </td><td>76</td></tr>\n",
       "\t<tr><td>Rare neurologic disease                       </td><td>68</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 4 × 2\n",
       "\\begin{tabular}{ll}\n",
       " clasificacion & n\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t Rare developmental defect during embryogenesis & 85\\\\\n",
       "\t Rare immune disease                            & 57\\\\\n",
       "\t Rare neoplastic disease                        & 76\\\\\n",
       "\t Rare neurologic disease                        & 68\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 4 × 2\n",
       "\n",
       "| clasificacion &lt;fct&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Rare developmental defect during embryogenesis | 85 |\n",
       "| Rare immune disease                            | 57 |\n",
       "| Rare neoplastic disease                        | 76 |\n",
       "| Rare neurologic disease                        | 68 |\n",
       "\n"
      ],
      "text/plain": [
       "  clasificacion                                  n \n",
       "1 Rare developmental defect during embryogenesis 85\n",
       "2 Rare immune disease                            57\n",
       "3 Rare neoplastic disease                        76\n",
       "4 Rare neurologic disease                        68"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "annot_ko_kstest_sign %>% group_by(clasificacion, sex, tissue, path) %>% summarise(n=n()) %>% \n",
    "group_by(clasificacion, sex, path) %>% summarise(n=n()) %>% filter(n==21) %>% \n",
    "group_by(clasificacion) %>% summarise(n=n()) %>% filter(n >40)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eca4edd2-2d87-4df7-922d-139d1b7c1b20",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
