{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7e0643c8-9b0f-4a3a-abf7-1c7361102686",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: igraph\n",
      "\n",
      "\n",
      "Attaching package: ‘igraph’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    decompose, spectrum\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "Loading required package: AnnotationHub\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    normalize, path, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: BiocFileCache\n",
      "\n",
      "Loading required package: dbplyr\n",
      "\n",
      "Loading required package: MultiAssayExperiment\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationHub’:\n",
      "\n",
      "    cache\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.18.0\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite either one:\n",
      "- Gu, Z. Complex Heatmap Visualization. iMeta 2022.\n",
      "- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "    genomic data. Bioinformatics 2016.\n",
      "\n",
      "\n",
      "The new InteractiveComplexHeatmap package can directly export static \n",
      "complex heatmaps into an interactive Shiny app with zero effort. Have a try!\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "\n",
      "\n",
      "========================================\n",
      "circlize version 0.4.15\n",
      "CRAN page: https://cran.r-project.org/package=circlize\n",
      "Github page: https://github.com/jokergoo/circlize\n",
      "Documentation: https://jokergoo.github.io/circlize_book/book/\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. circlize implements and enhances circular visualization\n",
      "  in R. Bioinformatics 2014.\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(circlize))\n",
      "========================================\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘circlize’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:igraph’:\n",
      "\n",
      "    degree\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘webshot’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    resize\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    resize\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    resize\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dbplyr’:\n",
      "\n",
      "    ident, sql\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:igraph’:\n",
      "\n",
      "    as_data_frame, groups, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(hipathia)\n",
    "library(AnnotationDbi)\n",
    "library(org.Hs.eg.db)\n",
    "library(ComplexHeatmap)\n",
    "library(circlize)\n",
    "library(ggplot2)\n",
    "library(jsonlite)\n",
    "library(networkD3)\n",
    "library(webshot)\n",
    "library(dplyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "57c290fd-c43c-4efd-a713-792e0ba4b674",
   "metadata": {},
   "outputs": [],
   "source": [
    "files_female <- list.files(\"/mnt/lustre/scratch/CBRA/research/projects/rrdd_mm/results/05_clasi_result_kstest/05_2_clasi_result_kstest_sign_only_female/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "37747751-6ff4-4639-b56c-b27165d66aae",
   "metadata": {},
   "outputs": [],
   "source": [
    "read_files <- function(x){\n",
    "    file <- paste(\"/mnt/lustre/scratch/CBRA/research/projects/rrdd_mm/results/05_clasi_result_kstest/05_2_clasi_result_kstest_sign_only_female/\", x,sep=\"\")\n",
    "    kstest_pv_adj_kovsctrl_sign <- readRDS(file)\n",
    "    kstest_pv_adj_kovsctrl_sign$ko <- rep(gsub(\".rds\",\"\",gsub(\"kstest_pvadj_logfc_kovsctrl_sign_only_female_\",\"\", x)), length(kstest_pv_adj_kovsctrl_sign$circ))\n",
    "    return(kstest_pv_adj_kovsctrl_sign)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "94ae032b-7f54-4238-a3f0-14a63bb198cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "ko_sign_female <- do.call(rbind,lapply(files_female, FUN=function(x) read_files(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "48227422-70d9-4f8e-95b1-93d918d1546f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loaded 146 pathways\n",
      "\n"
     ]
    }
   ],
   "source": [
    "pathways <- load_pathways(\"hsa\")\n",
    "ko_sign_female$hip_name <- get_path_names(pathways, ko_sign_female$circ)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f9c7c45f-9142-4d21-986d-80728cc2fd56",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned many:1 mapping between keys and columns\n",
      "\n"
     ]
    }
   ],
   "source": [
    "name_enfermerdad_symbol <- readRDS(\"/mnt/lustre/scratch/CBRA/research/projects/rrdd_mm/results/00_parser_orphanet/gen_enfermedad/name_enfermerdad_symbol.rds\")\n",
    "colnames(name_enfermerdad_symbol) <- c(\"name_enf\",\"gene\")\n",
    "symbol_entreid <- AnnotationDbi::select(org.Hs.eg.db, keys=paste(name_enfermerdad_symbol$gene), columns=c(\"ENTREZID\"),keytype=\"SYMBOL\")\n",
    "\n",
    "gen_enf <- unique(merge(name_enfermerdad_symbol, symbol_entreid, by.x=\"gene\",by.y=\"SYMBOL\"))\n",
    "rd_gen_class <- readRDS(\"/mnt/lustre/scratch/CBRA/research/projects/rrdd_mm/results/00_parser_orphanet/rd_gen_class.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9aaacbaa-a080-48d1-bbea-b59eecaa3533",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>enfermedad</th><th scope=col>clasificacion</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>6884</th><td>STAG1</td><td>STAG1-related intellectual disability-facial dysmorphism-gastroesophageal reflux syndrome</td><td>Rare developmental defect during embryogenesis</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & V1 & enfermedad & clasificacion\\\\\n",
       "  & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t6884 & STAG1 & STAG1-related intellectual disability-facial dysmorphism-gastroesophageal reflux syndrome & Rare developmental defect during embryogenesis\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 3\n",
       "\n",
       "| <!--/--> | V1 &lt;fct&gt; | enfermedad &lt;fct&gt; | clasificacion &lt;fct&gt; |\n",
       "|---|---|---|---|\n",
       "| 6884 | STAG1 | STAG1-related intellectual disability-facial dysmorphism-gastroesophageal reflux syndrome | Rare developmental defect during embryogenesis |\n",
       "\n"
      ],
      "text/plain": [
       "     V1   \n",
       "6884 STAG1\n",
       "     enfermedad                                                                               \n",
       "6884 STAG1-related intellectual disability-facial dysmorphism-gastroesophageal reflux syndrome\n",
       "     clasificacion                                 \n",
       "6884 Rare developmental defect during embryogenesis"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>enfermedad</th><th scope=col>clasificacion</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>SOX5 </td><td>12p12.1 microdeletion syndrome    </td><td>Rare developmental defect during embryogenesis</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>LEMD3</td><td>12q14 microdeletion syndrome      </td><td>Rare bone disease                             </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>CNOT2</td><td>12q15q21.1 microdeletion syndrome </td><td>Rare developmental defect during embryogenesis</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>FOXG1</td><td>14q11.2 microduplication syndrome </td><td>Rare developmental defect during embryogenesis</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>FOXG1</td><td>14q12 microdeletion syndrome      </td><td>Rare developmental defect during embryogenesis</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>UBE3A</td><td>15q11q13 microduplication syndrome</td><td>Rare developmental defect during embryogenesis</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & V1 & enfermedad & clasificacion\\\\\n",
       "  & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t1 & SOX5  & 12p12.1 microdeletion syndrome     & Rare developmental defect during embryogenesis\\\\\n",
       "\t3 & LEMD3 & 12q14 microdeletion syndrome       & Rare bone disease                             \\\\\n",
       "\t5 & CNOT2 & 12q15q21.1 microdeletion syndrome  & Rare developmental defect during embryogenesis\\\\\n",
       "\t7 & FOXG1 & 14q11.2 microduplication syndrome  & Rare developmental defect during embryogenesis\\\\\n",
       "\t9 & FOXG1 & 14q12 microdeletion syndrome       & Rare developmental defect during embryogenesis\\\\\n",
       "\t11 & UBE3A & 15q11q13 microduplication syndrome & Rare developmental defect during embryogenesis\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | V1 &lt;fct&gt; | enfermedad &lt;fct&gt; | clasificacion &lt;fct&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | SOX5  | 12p12.1 microdeletion syndrome     | Rare developmental defect during embryogenesis |\n",
       "| 3 | LEMD3 | 12q14 microdeletion syndrome       | Rare bone disease                              |\n",
       "| 5 | CNOT2 | 12q15q21.1 microdeletion syndrome  | Rare developmental defect during embryogenesis |\n",
       "| 7 | FOXG1 | 14q11.2 microduplication syndrome  | Rare developmental defect during embryogenesis |\n",
       "| 9 | FOXG1 | 14q12 microdeletion syndrome       | Rare developmental defect during embryogenesis |\n",
       "| 11 | UBE3A | 15q11q13 microduplication syndrome | Rare developmental defect during embryogenesis |\n",
       "\n"
      ],
      "text/plain": [
       "   V1    enfermedad                        \n",
       "1  SOX5  12p12.1 microdeletion syndrome    \n",
       "3  LEMD3 12q14 microdeletion syndrome      \n",
       "5  CNOT2 12q15q21.1 microdeletion syndrome \n",
       "7  FOXG1 14q11.2 microduplication syndrome \n",
       "9  FOXG1 14q12 microdeletion syndrome      \n",
       "11 UBE3A 15q11q13 microduplication syndrome\n",
       "   clasificacion                                 \n",
       "1  Rare developmental defect during embryogenesis\n",
       "3  Rare bone disease                             \n",
       "5  Rare developmental defect during embryogenesis\n",
       "7  Rare developmental defect during embryogenesis\n",
       "9  Rare developmental defect during embryogenesis\n",
       "11 Rare developmental defect during embryogenesis"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>kspvadj</th><th scope=col>logFC</th><th scope=col>ko</th><th scope=col>circ</th><th scope=col>sex</th><th scope=col>tissue</th><th scope=col>hip_name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>4</th><td>0.01314045</td><td>-0.09978966</td><td>10274</td><td>P-hsa04110-43 83 84 85</td><td>female</td><td>Heart</td><td>Cell cycle: SMC1B SMC3 STAG1 RAD21</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 7\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & kspvadj & logFC & ko & circ & sex & tissue & hip\\_name\\\\\n",
       "  & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t4 & 0.01314045 & -0.09978966 & 10274 & P-hsa04110-43 83 84 85 & female & Heart & Cell cycle: SMC1B SMC3 STAG1 RAD21\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 7\n",
       "\n",
       "| <!--/--> | kspvadj &lt;dbl&gt; | logFC &lt;dbl&gt; | ko &lt;chr&gt; | circ &lt;chr&gt; | sex &lt;chr&gt; | tissue &lt;chr&gt; | hip_name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| 4 | 0.01314045 | -0.09978966 | 10274 | P-hsa04110-43 83 84 85 | female | Heart | Cell cycle: SMC1B SMC3 STAG1 RAD21 |\n",
       "\n"
      ],
      "text/plain": [
       "  kspvadj    logFC       ko    circ                   sex    tissue\n",
       "4 0.01314045 -0.09978966 10274 P-hsa04110-43 83 84 85 female Heart \n",
       "  hip_name                          \n",
       "4 Cell cycle: SMC1B SMC3 STAG1 RAD21"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rd_gen_class[which(rd_gen_class$V1 == paste(gen_enf[which(gen_enf$ENTREZID == ko_sign_female$ko),\"gene\"])),]\n",
    "head(rd_gen_class)\n",
    "ko_sign_female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e64d3b1-7e40-4451-aa73-7e4a240c5f40",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
